### PR TITLE
`Programming exercises`: Fix an issue with hidden feedback for teaching assistants

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.

### Motivation and Context

#5765 reported an inconsistent behavior regarding hiding and showing the feedback of tests after the due date to TAs before the due date.

### Description

This PR fixes the filtering of these feedbacks to always show the hidden feedback to keep the behavior consistent across Artemis.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise with Tests after the due date

1. Log in to Artemis
2. Participate in the exercise as tutor/instructor and check the feedbacks right after receiving the results on Artemis
3. Reload the page and check again if you can still see the feedback of the hidden test cases

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java [hunk 1]
```diff
@@ -152,7 +152,9 @@ public boolean canAccessParticipation(@NotNull ProgrammingExerciseParticipation
         if (participation instanceof ProgrammingExerciseStudentParticipation studentParticipation && studentParticipation.isOwnedBy(user)) {
             return true;
         }
-        final var programmingExercise = programmingExerciseRepository.getProgrammingExerciseFromParticipation(participation);
+
+        ProgrammingExercise programmingExercise = participation.getProgrammingExercise() != null ? participation.getProgrammingExercise()
+                : programmingExerciseRepository.getProgrammingExerciseFromParticipation(participation);
```
Comment (b-fein) [inline src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java:157]: Isn’t this basically what `getProgrammingExerciseFromParticipation` does internally as well? Why is it needed now?

Comment (JohannesStoehr) [reply src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java]: Oops this somehow slipped through. I reverted that change

File: src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java [hunk 2]
```diff
@@ -76,10 +77,7 @@ private void checkIsAtLeastEditorInCourseElseThrow(@NotNull Course course, @Null
      * @return true if the passed user is at least an editor in the course, false otherwise
      */
     public boolean isAtLeastEditorInCourse(@NotNull Course course, @Nullable User user) {
-        if (user == null || user.getGroups() == null) {
-            // only retrieve the user and the groups if the user is null or the groups are missing (to save performance)
-            user = userRepository.getUserWithGroupsAndAuthorities();
-        }
+        user = loadUserIfNeeded(user);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java:80]: If someone has a suggestion on how to write this nicely without a warning, feel free to add your suggestions.
Simply adding a new variable does not really seam helpful to me in this case.

File: src/main/java/de/tum/in/www1/artemis/web/rest/QuizSubmissionResource.java [hunk 3]
```diff
@@ -155,9 +155,10 @@ public ResponseEntity<Result> submitForPractice(@PathVariable Long exerciseId, @
         }
 
         quizExercise.setQuizPointStatistic(null);
-        quizExercise.setCourse(null);
 
         messagingService.broadcastNewResult(result.getParticipation(), result);
+
+        quizExercise.setCourse(null);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/web/rest/QuizSubmissionResource.java:161]: Why is this change necessary?

Comment (JohannesStoehr) [reply src/main/java/de/tum/in/www1/artemis/web/rest/QuizSubmissionResource.java:161]: Otherwise we cannot determine if the feedbacks have to be filtered before broadcasting

File: src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java (hunk 2 above)
Comment (b-fein) [reply src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java:80]: I think we can ignore the suggestion in this case, as in the current implementation the callers depend on it that the user parameter gets replaced with the user with groups.

**Review by stefanwaldhauser — APPROVED** [review]
LGTM :) Gj

**Review by b-fein — APPROVED** [review]
Manual testing: works as intended. :+1:
